.github/actions/build.bash

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```

Comment:
If you wanna write a js github action, you need to include your node_modules in git. `ncc` seems to be the recommend way of bundling it all together

Reply:
What's the benefit of creating these custom actions as opposed to using existing ones?

Reply:
It seems to be the best way (that I've found so far) to keep things DRYish. For example, the release publish job is identical for haskell/rust. What I had using marketplace actions was roughly -

```yaml
  publish:
    if: startsWith(github.ref, 'refs/tags/v')
    needs:
      - build
    name: Publish github release
    runs-on: ubuntu-latest
    steps:
       - uses: actions/download-artifact@v2
         with:
           path: ./artifacts

       - name: Create Release
         id: create_release
         uses: actions/create-release@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           tag_name: ${{ github.event.ref }}
           release_name: ${{ github.event.ref }}
      
       - name: Upload artifacts
         uses: csexton/release-asset-action@v2
         with:
           pattern: |
             artifacts/**/*@(.zip|.tar.gz)
           github-token: ${{ secrets.GITHUB_TOKEN }}
           release-url: ${{ steps.create_release.outputs.upload_url }}
```

and I wanted to figure out a way to avoid copy-pasting this into `haskell.yaml` and `rust.yaml`. 

There is this concept of [composite steps](https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action), which in theory let you create new actions from a sequence of steps. The problem is you can't call other actions inside composite actions, you can only use `run` (i.e. script) steps. It's really odd, kind of like a language that lets you define functions, but only if they don't contain other non built-in functions. 

There is actually an issue about it here - https://github.com/actions/runner/issues/646 (what I'm referring to is the lack of `uses` support)

That all being said, not actually that against the copy/paste approach. The js actions are kind of a lot of overhead for not much functionality

Reply:
> That all being said, not actually that against the copy/paste approach. The js actions are kind of a lot of overhead for not much functionality

Being DRY would be good, but yeah we need to consider the complexity/understandability trade-off.  Another way we could be DRY-ish is to trigger another workflow that has all of the things it acts on abstracted into variables.  The github release publish is an example:

https://github.com/swift-nav/gnss-converters-private/pull/625/files#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R210

If it's possible trigger a workflow that lives in another yaml file, maybe we could achieve some abstraction by having a library of abstract/parameterized workflows.  IOW, we'd be using workflows as our means of abstraction, rather than actions.